generated/nidaqmx/task.py

```diff
@@ -343,7 +341,22 @@ def close(self):
                 'Attempted to close NI-DAQmx task of name "{}" but task was '
                 'already closed.'.format(self._saved_name), DaqResourceWarning)
             return
-        
+
+        if self._done_event_stack is not None or self._every_n_acquired_event_stack is not None:
+            # Abort the task so it stops calling event callbacks.
+            self.control(TaskMode.TASK_ABORT)
+
+            # Clean up event handlers. For LibraryInterpreter, this destroys the ctypes callback
+            # pointer. For GrpcStubInterpreter, this cancels the response stream and joins the
+            # event thread, which ensures that the thread doesn't call event callbacks.
+            if self._every_n_acquired_event_stack is not None:
+                self._every_n_acquired_event_stack.close()
+                self._every_n_acquired_event_stack = None
+
+            if self._done_event_stack is not None:
+                self._done_event_stack.close()
+                self._done_event_stack = None
+
         self._interpreter.clear_task(self._handle)
 
```

Comment:
It's probably better to split cancel vs. close so that we can cancel events, clear task, and close events.